src/python/gudhi/sktda/vector_methods.py

```diff
@@ -0,0 +1,481 @@
+        """
+        self.threshold = threshold
+
+    def fit(self, X, y=None):
+        """
+        Fit the TopologicalVector class on a list of persistence diagrams (this function actually does nothing but is useful when TopologicalVector is included in a scikit-learn Pipeline).
+
+        Parameters:
+            X (list of n x 2 or n x 1 numpy arrays): input persistence diagrams.
+            y (n x 1 array): persistence diagram labels (unused).
+        """
+        return self
+
+    def transform(self, X):
+        """
+        Compute the topological vector for each persistence diagram individually and concatenate the results.
+
+        Parameters:
+            X (list of n x 2 numpy arrays): input persistence diagrams.
+    
+        Returns:
+            Xfit (numpy array with shape (number of diagrams) x (**threshold**): output topological vectors.
+        """
+        if self.threshold == -1:
+            thresh = np.array([X[i].shape[0] for i in range(len(X))]).max()
+        else:
+            thresh = self.threshold
+
+        num_diag = len(X)
+        Xfit = np.zeros([num_diag, thresh])
+
+        for i in range(num_diag):
+
+            diagram, num_pts_in_diag = X[i], X[i].shape[0]
+            pers = 0.5 * (diagram[:,1]-diagram[:,0])
+            min_pers = np.minimum(pers,np.transpose(pers))
+            distances = DistanceMetric.get_metric("chebyshev").pairwise(diagram)
+            vect = np.flip(np.sort(np.triu(np.minimum(distances, min_pers)), axis=None), 0)
+            dim = min(len(vect), thresh)
+            Xfit[i, :dim] = vect[:dim]
+
+        return Xfit
+
+class ComplexPolynomial(BaseEstimator, TransformerMixin):
+    """
+    This is a class for computing complex polynomials from a list of persistence diagrams. The persistence diagram points are seen as the roots of some complex polynomial, whose coefficients are returned in a complex vector. See https://link.springer.com/chapter/10.1007%2F978-3-319-23231-7_27 for more details.
+    """
+    def __init__(self, F="R", threshold=10):
+        """
+        Constructor for the ComplexPolynomial class.
+
+        Attributes:
+           F (char): either "R", "S" or "T" (default "R"). Type of complex polynomial that is going to be computed (explained in https://link.springer.com/chapter/10.1007%2F978-3-319-23231-7_27).
+           threshold (int): number of coefficients (default 10). This is the dimension of the complex vector of coefficients. If -1, this threshold is computed from the list of persistence diagrams by considering the one with the largest number of points and using the dimension of its corresponding complex vector of coefficients as threshold. 
```

Comment:
Do you keep the low coefficients or the high coefficients?

Reply:
High coefficients. Doc should be more explicit now.